redox/response_flowsheet.go

```diff
@@ -252,37 +243,79 @@ func PopulateBGMObservations(stats *clinics.BgmStatsV1, settings FlowsheetSettin
 		}
 	}
 
+	unitsDay := day
+	unitsPercentage := percentage
+	sourceGlucoseUnits := string(clinics.MmolL)
+	destGlucoseUnits := settings.PreferredBGUnits
+
 	var averageDailyRecords *float64
 	var averageGlucose *float64
-	var averageGlucoseUnits *string
 	var timeInVeryLowRecords *int
 	var timeInVeryHighRecords *int
+	var timeInVeryLowPercent *float64
+	var timeInLowPercent *float64
+	var timeInTargetPercent *float64
+	var timeInHighPercent *float64
+	var timeInVeryHighPercent *float64
+	var bgmStdDev *float64
+	var bgmCoeffVar *float64
+	var bgmDaysWithData *int
+	var bgmTotalRecords *int
+	var minGlucose *float64
+	var maxGlucose *float64
 
 	if period != nil {
 		if period.AverageGlucoseMmol != nil {
-			val := *period.AverageGlucoseMmol
-			units := string(clinics.MmolL)
+			// Convert blood glucose to preferred units, store unit result overriding preference if unit is not convertable
+			averageGlucoseVal, units := bgInUnits(float64(*period.AverageGlucoseMmol), sourceGlucoseUnits, destGlucoseUnits)
+			averageGlucose = &averageGlucoseVal
+			destGlucoseUnits = units
+		}
 
-			// Convert blood glucose to preferred units
-			val, units = bgInUnits(val, units, settings.PreferredBGUnits)
+		// Convert standard deviation to preferred units
+		bgmStdDevVal, _ := bgInUnits(period.StandardDeviation, sourceGlucoseUnits, destGlucoseUnits)
+		bgmStdDev = &bgmStdDevVal
 
-			averageGlucose = &val
-			averageGlucoseUnits = &units
-		}
-		averageDailyRecords = period.AverageDailyRecords
+		// Convert min glucose to preferred units
+		minGlucoseVal, _ := bgInUnits(period.Min, sourceGlucoseUnits, destGlucoseUnits)
+		minGlucose = &minGlucoseVal
 
+		// Convert max glucose to preferred units
+		maxGlucoseVal, _ := bgInUnits(period.Max, sourceGlucoseUnits, destGlucoseUnits)
+		maxGlucose = &maxGlucoseVal
+
+		averageDailyRecords = period.AverageDailyRecords
 		timeInVeryLowRecords = period.TimeInVeryLowRecords
 		timeInVeryHighRecords = period.TimeInVeryHighRecords
+		bgmCoeffVar = &period.CoefficientOfVariation
+		bgmDaysWithData = &period.DaysWithData
+		bgmTotalRecords = period.TotalRecords
+		timeInVeryLowPercent = period.TimeInVeryLowPercent
+		timeInLowPercent = period.TimeInLowPercent
+		timeInTargetPercent = period.TimeInTargetPercent
+		timeInHighPercent = period.TimeInHighPercent
+		timeInVeryHighPercent = period.TimeInVeryHighPercent
 	}
 
 	observations := []*Observation{
 		{"REPORTING_PERIOD_START_SMBG", formatTime(periodStart), "DateTime", nil, reportingTime, "SMBG Reporting Period Start"},
 		{"REPORTING_PERIOD_END_SMBG", formatTime(periodEnd), "DateTime", nil, reportingTime, "SMBG Reporting Period End"},
 		{"REPORTING_PERIOD_START_SMBG_DATA", formatTime(firstData), "DateTime", nil, reportingTime, "SMBG Reporting Period Start Date of actual Data"},
+		{"TIME_ABOVE_RANGE_VERY_HIGH_SMBG", formatFloat(unitIntervalToPercent(timeInVeryHighPercent)), "Numeric", &unitsPercentage, reportingTime, "% of readings > 250 mg/dL (>13.9 mmol/L)"},
```

Comment:
These names would be better if they included PERCENT, RATIO, or PCT, since they're representing a percentage and not a duration.